pkg/os/shell/shell_windows.go

```diff
@@ -58,6 +58,19 @@ func getNameAndItsPpid(pid uint32) (exefile string, parentid uint32, err error)
 func detect() (string, error) {
 	shell := os.Getenv("SHELL")
 
+	baseShell := filepath.Base(shell)
+
+	if baseShell != "powershell" && baseShell != "cmd" {
+		switch {
+		case strings.Contains(strings.ToLower(shell), "powershell"):
+			return "powershell", nil
+		case strings.Contains(strings.ToLower(shell), "pwsh"):
+			return "powershell", nil
+		case strings.Contains(strings.ToLower(shell), "cmd"):
+			return "cmd", nil
+		}
```

Comment:
This is duplication of https://github.com/crc-org/crc/pull/4401/files#diff-ccde05d43b105715d5a716447bc74186b326dc1d90d98091e780bb407da6fd3fR84-R89 better to have a helper function which take the `shell` as argument and return type.

Also this logic should be at the end instead of start.